lib/div-alg.gi

```diff
@@ -2309,33 +2309,36 @@ for i in [1..k] do
 od;
 
 return [n,F,m,hrs,c];
-
 end);
 
+
 ################################################
 # IsCyclotomicExtension checks whether the input are two number fields 
 # and the first is a cyclotomic extension of the second
 ################################################
 
 InstallGlobalFunction( IsCyclotomicExtension, function(K,F)
-#IsCyclotomicExtension := function(K,F)
 
-local c,pr,i;
+local c,pr,e;
 
-if not IsNumberField(K) or not IsNumberField(F) or not IsSubset(K,F) then 
-  return fail;
+if not IsAbelianNumberField(K) or not IsAbelianNumberField(F) or not IsSubset(K,F) then 
+  return 0;
```

Comment:
In GAP, there is a convention that `IsSomething` returns a boolean. Do we need to change it to return a number? Is seems from it's use that the numerical output is not utilised anywhere. If you think that you may need it, I suggest to have a separate function for it.

Reply:
> In GAP, there is a convention that `IsSomething` returns a boolean. Do we need to change it to return a number? Is seems from it's use that the numerical output is not utilised anywhere. If you think that you may need it, I suggest to have a separate function for it.

I changed the output of IsCyclotomicExtension to a number because this number is used in ReducingCyclotomicExtension. I changed that to deal with the issue mentioned by Allen yesterday. One can have a separate function but the boolean function has not additional use and the cost of both is the same, so I guess that the most efficent is simply to change the name to CyclotomicExtensionGenerator of something like that on which one can replace the output 0, which is the case where the extension is not really cyclotomic, by fail.

Reply:
I agree with `CyclotomicExtensionGenerator` with `fail` instead of zero. Internal function, not documented, fine to change the semantics then.